public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -664,3 +671,32 @@ function MaybeQueryLibrarySaveButton(props: { query: DataQuery }) {
   const { renderSaveQueryButton } = useQueryLibraryContext();
   return renderSaveQueryButton(props.query);
 }
+
+function AdaptiveTelemetryQueryActions({ query }: { query: DataQuery }) {
+  const extensionPointId = 'grafana/adaptivetelemetry/query/action';
+
+  type AdaptiveTelemetryQueryActionProps = {
+    /** An ordered list of lower-case [a-z]+ string identifiers to provide context clues of where this component is being embedded and how we might want to consider displaying it */
+    contextHints?: string[];
+    query?: DataQuery;
+  };
```

Comment:
Maybe worth adding this one to `@grafana/data` as well. Then you can use that in the apps consuming this extension point.